roaring/roaring.go

```diff
@@ -1267,12 +1267,33 @@ func (c *container) bitmapToRun() {
 	// TODO
 }
 
+// arrayToRun converts from array format to RLE format
 func (c *container) arrayToRun() {
-	// TODO
+	c.runs = make([]interval32, 0, c.n) // what capacity to use?
+	start := c.array[0]
+	for i, v := range c.array[1:] {
+		if v - c.array[i] > 1 {
+			// if current-previous > 1, one run ends and another begins
+			c.runs = append(c.runs, interval32{start, c.array[i]})
+			start = v
+		}
+	}
+	// append final run
+	c.runs = append(c.runs, interval32{start, c.array[c.n-1]})
```

Comment:
I would prefer c.array[len(c.array)-1].  Though I guess we should trust c.n to be accurate.